### PR TITLE
[HOLD] move from OpenShiftTestAssistant to Arquillian Cube

### PR DESCRIPTION
**Pull request by Ladicek**

### CODE REVIEW

**Comment by Ladicek** [comment]
I also reformatted the `pom.xml`s to use 2-space indentation (because that seems to be what the original `pom.xml` used). This clutters the diff somewhat; append `?w=1` to the URL to see the diff without whitespace changes.

**Comment by Ladicek** [comment]
Please hold this, because I found an issue. After the test, pods keep hanging in the "Terminating" state, which is super bad.

EDIT: actually, scratch this. This only happens with Minishift and the Minishift CentOS ISO 1.1.0; other OpenShift flavors (OCP, oc cluster up, Minishift with the boot2docker ISO) aren't affected. Updating to latest Minishift (1.6.0) and latest Minishift CentOS ISO (1.2.0) fixes the issue.

File: pom.xml [hunk 1]
```diff
@@ -87,11 +96,34 @@
...
   </dependencies>
 
   <build>
     <plugins>
       <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <classpathDependencyExcludes>
+            <classpathDependencyExclude>org.arquillian.cube:arquillian-cube-openshift</classpathDependencyExclude>
```
Comment (cescoffier) [inline pom.xml:120]: why is this required?

File: src/test/java/io/openshift/booster/OpenshiftIT.java [hunk 2]
```diff
@@ -18,51 +18,31 @@
...
 
-    @BeforeClass
-    public static void setup() throws Exception {
-        // the application itself
-        openshift.deployApplication();
-
-        // wait until the pods & routes become available
-        openshift.awaitApplicationReadinessOrFail();
-
-        await().atMost(5, TimeUnit.MINUTES).until(() -> {
```
Comment (cescoffier) [inline src/test/java/io/openshift/booster/OpenshiftIT.java]: My experience has shown that it's better to test the response and not wait only for the "readiness". Something things are wild for a few seconds.

Comment (cescoffier) [reply src/test/java/io/openshift/booster/OpenshiftIT.java]: (this is why we have the 2 checks BTW)

File: src/test/resources/arquillian.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<arquillian xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns="http://jboss.org/schema/arquillian"
+            xsi:schemaLocation="http://jboss.org/schema/arquillian http://jboss.org/schema/arquillian/arquillian_1_0.xsd">
+
+  <extension qualifier="openshift">
+    <property name="namespace.use.current">true</property>
```
Comment (cescoffier) [inline src/test/resources/arquillian.xml:6]: So basically, we cannot use an ephemeral project. We force it to be the current one.

File: pom.xml (hunk 1 above)
Comment (Ladicek) [reply pom.xml:120]: Because Swarm uses Arquillian for unit tests as well. Without these excludes, Arquillian Cube would fire in the unit tests phase, which 1. is just wrong, 2. doesn't work because the image stream doesn't exist yet (`fabric8:build` wasn't executed yet), so Cube will wait indefinitely for the pod to be started.

File: src/test/java/io/openshift/booster/OpenshiftIT.java (hunk 2 above)
Comment (Ladicek) [reply src/test/java/io/openshift/booster/OpenshiftIT.java]: Cube just awaits readiness, true. Could add that back, but I didn't see a failure yet. Obviously, everything works fine locally...

File: src/test/resources/arquillian.xml (hunk 3 above)
Comment (Ladicek) [reply src/test/resources/arquillian.xml:6]: We can. I added this so that the test behavior doesn't change, but if I remove this line and enable image stream detection, the test runs in `itest-xxxx` namespace just fine.

I thought I could force `-Dnamespace.use.current=false` on the cmdline, but it seems Arquillian has the priorities backwards :-( (EDIT: I filed https://github.com/arquillian/arquillian-cube/issues/816 for this.)

**Comment by Ladicek** [comment]
Rebased and added back the awaiting of the route to return 200.

**Comment by Ladicek** [comment]
Updated to Arquillian Cube 1.9.1.

File: src/test/resources/arquillian.xml (hunk 3 above)
Comment (Ladicek) [reply src/test/resources/arquillian.xml:6]: I now updated to Cube 1.9.1 and using an ephemeral project is now easily possible.

`mvn clean verify -Popenshift,openshift-it` runs the tests in current project.

`mvn clean verify -Popenshift,openshift-it -Dnamespace.use.current=false -DenableImageStreamDetection=true` runs the tests in an ephemeral project.

**Review by kenfinnigan — APPROVED** [review]
